### PR TITLE
Support unix_ms_now in rhai engine

### LINKED ISSUE #5182: Support unix time in ms in rhai scripts
**Is your feature request related to a problem? Please describe.**
We would like to measure precisie timings using the coprocessor, but it's currently impossible since the only available timing function is `unix_now` which returns seconds since epoch.

**Describe the solution you'd like**
a new function `unix_ms_now` that returns exact ms of current time.

### PR DESCRIPTION
**Pull request by shaikatzz**
Fixes #5182 

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/plugins/rhai/tests.rs [hunk 1]
```diff
@@ -579,6 +579,21 @@ fn it_can_create_unix_now() {
...
+    let engine = new_rhai_test_engine();
+    let st = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("can get system time")
+        .as_millis() as i64;
+    let unix_ms_now: i64 = engine
+        .eval(r#"unix_ms_now()"#)
+        .expect("can get unix_ms_now() timestamp");
+    // Always difficult to do timing tests. unix_ms_now() should execute within a second of st,
+    // so...
```
Comment (garypen) [inline apollo-router/src/plugins/rhai/tests.rs:593]: :)

Perhaps reverse the ordering of the timestamp captures and assert that st >= unix_ms_now ?

**Review by garypen — COMMENTED** [review]
Thanks for this PR and for the excellent presentation. I particularly appreciate the test and the docs additions.

Could you also create a changeset (so you get appropriate credit when this is released) following the procedure in `.changesets/README.md` ? re-request a review from me when you are ready.

File: apollo-router/src/plugins/rhai/tests.rs (hunk 1 above)
Comment (shaikatzz) [reply apollo-router/src/plugins/rhai/tests.rs:593]: Actually just followed the same pattern in the test above `it_can_create_unix_now`.
Why does the ordering matter? anyway we will want to check that the second capture ran within a second of the first one, to see that the results makes sense.

**Comment by shaikatzz** [comment]
> Thanks for this PR and for the excellent presentation. I particularly appreciate the test and the docs additions.
> 
> Could you also create a changeset (so you get appropriate credit when this is released) following the procedure in `.changesets/README.md` ? re-request a review from me when you are ready.

Done!

File: apollo-router/src/plugins/rhai/tests.rs (hunk 1 above)
Comment (garypen) [reply apollo-router/src/plugins/rhai/tests.rs:593]: It's all good and, as you observe the previous test should be re-ordered.

I was basically thinking that if we re-ordered the samples, then we'd just need to assert that the second timestamp was `>=` the first one. That would eliminate the flakiness of guesstimating that it should run within a second.

Anyway, ignore this for now and we'll change it if the test ever becomes flaky.

**Review by garypen — APPROVED** [review]

**Comment by shaikatzz** [comment]
@IvanGoncharov @garypen do you know why the tests are not passing? what else is required in order to merge that PR?

**Comment by garypen** [comment]
> @IvanGoncharov @garypen do you know why the tests are not passing? what else is required in order to merge that PR?

possibly because the CLA wasn't signed. I've just kicked it back into life and we'll see what happens after this run.

**Comment by shaikatzz** [comment]
@garypen I've signed the CLA, let me know if anything else is needed from my side, i'll be happy to have this change merged :-)

**Pull request merged**